src/svg/axis.js

```diff
@@ -0,0 +1,211 @@
+              tickArguments = [10],
+              tickValues = null,
+              tickFormat = null,
+              outerTickSize = 6,
+              innerTickSize = 6,
+              tickPadding = 3,
+              svgDomainLine = d3.svg.line();
+
+        var dataJoin = fc.util.dataJoin()
+            .selector('g.tick')
+            .element('g')
+            .key(fc.util.fn.identity)
+            .attrs({'class': 'tick'});
+
+        // returns a function that creates a translation based on
+        // the bound data
+        function containerTranslate(s, trans) {
+            return function(d) {
+                return trans(s(d), 0);
+            };
+        }
+
+        function translate(x, y) {
+            return 'translate(' + x + ', ' + y + ')';
+        }
+
+        function translateTransposed(x, y) {
+            return 'translate(' + y + ', ' + x + ')';
+        }
+
+        function transposeArray(arr) {
+            return arr.map(function(d) {
+                return [d[1], d[0]];
+            });
+        }
+
+        function isVertical() {
+            return orient === 'left' || orient === 'right';
+        }
+
+        var axis = function(selection) {
+
+            selection.each(function(data, index) {
+
+                // Stash a snapshot of the new scale, and retrieve the old snapshot.
+                var scaleOld = this.__chart__ || scale;
+                this.__chart__ = scale.copy();
+
+                function tryApply(fn, defaultVal) {
+                    return scale[fn] ? scale[fn].apply(scale, tickArguments) : defaultVal;
+                }
+
+                var ticksArray = tickValues == null ? tryApply('ticks', scale.domain()) : tickValues;
+                var tickFormatter = tickFormat == null ? tryApply('tickFormat', fc.util.fn.identity) : tickFormat;
+
+                // orientation logic
+                var sign = orient === 'bottom' || orient === 'right' ? 1 : -1;
+                var pathTranspose = isVertical() ? transposeArray : fc.util.fn.identity;
+                var transform = isVertical() ? translateTransposed : translate;
+
+                var container = d3.select(this);
+
+                // add the domain line
+                var range = fc.util.scale.range(scale);
+                var domainPathData = pathTranspose([
+                      [range[0], sign * outerTickSize],
+                      [range[0], 0],
+                      [range[1], 0],
+                      [range[1], sign * outerTickSize]
+                    ]);
+
+                var domainLine = container.selectAll('path.domain')
+                    .data([data]);
+
+                domainLine.enter()
+                    .append('path')
+                    .classed('domain', true);
+
+                d3.transition(domainLine)
+                    .attr('d', svgDomainLine(domainPathData));
+
+                // datajoin and construct the ticks / label
+                dataJoin.attrs({
```

Comment:
I don't think this will work if you change the orientation, the attributes will only be set on enter.